pkg/api/pod/util.go

```diff
@@ -518,6 +518,24 @@ func dropDisabledFields(
 		}
 		// For other types of containers, validateContainers will handle them.
 	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodLifecycleSleepAction) && !podLifecycleSleepActionInUse(oldPodSpec) {
+		for i := range podSpec.Containers {
+			if podSpec.Containers[i].Lifecycle != nil && podSpec.Containers[i].Lifecycle.PreStop != nil {
+				podSpec.Containers[i].Lifecycle.PreStop.Sleep = nil
```

Comment:
You need to handle it in `PostStart` too.  It's not super meaningful (actually, might be useful for sidecar hacks) but the same type is used for both hooks.